File: windows/tests/ConnectorControl.Core.Tests/State/PathContextTests.cs
using ConnectorControl.Core.State;

namespace ConnectorControl.Core.Tests.State;

public class PathContextTests
{
    [Fact]
    public void LiveReadsTheProcessEnvironmentAndKnownFolders()
    {
        Environment.SetEnvironmentVariable("CONNECTOR_CONTROL_PLAN_PROBE", "yes");
        try
        {
            var live = PathContext.Live();
            Assert.Equal("yes", live.Environment["CONNECTOR_CONTROL_PLAN_PROBE"]);
            Assert.Equal(KnownFolders.Current(), live.Folders);
            Assert.IsType<RealPathProbe>(live.Probe);
        }
        finally
        {
            Environment.SetEnvironmentVariable("CONNECTOR_CONTROL_PLAN_PROBE", null);
        }
    }
}

File: Tests/ConnectorControlStateTests/PathContextTests.swift
import XCTest
@testable import ConnectorControlState

/// Mirror: windows/tests/ConnectorControl.Core.Tests/State/PathContextTests.cs
final class PathContextTests: XCTestCase {
    func testLiveReadsTheProcessEnvironmentAndApplicationSupport() {
        setenv("CONNECTOR_CONTROL_PLAN_PROBE", "yes", 1)
        defer { unsetenv("CONNECTOR_CONTROL_PLAN_PROBE") }
        let live = PathContext.live()
        XCTAssertEqual(live.environment["CONNECTOR_CONTROL_PLAN_PROBE"], "yes")
        XCTAssertEqual(live.appSupport,
                       FileManager.default.homeDirectoryForCurrentUser
                           .appendingPathComponent("Library/Application Support"))
    }
}
